iroha/src/tx.rs

```diff
@@ -111,12 +116,18 @@ impl Transaction {
         }
     }
 
-    pub fn validate(self) -> Result<Transaction, String> {
+    pub fn validate(self, world_state_view: &WorldStateView) -> Result<Transaction, String> {
         if let Transaction::Signed {
             request,
             signatures,
         } = self
         {
+            let mut world_state_view = world_state_view.clone();
+            for instruction in &request.instructions {
+                if let Err(e) = instruction.invoke(&mut world_state_view) {
+                    return Err(e);
+                }
```

Comment:
Can't we use `instruction.invoke(&mut world_state_view)?;` instead of `if let`?

Reply:
You are right!